mutation/mutation_partition.cc

```diff
@@ -479,6 +481,21 @@ mutation_partition::apply(const schema& s, mutation_partition_view p,
     apply_monotonically(s, std::move(p2), no_cache_tracker, app_stats, is_preemptible::no, res);
 }
 
+future<>
+mutation_partition::apply_gently(const schema& s, mutation_partition_view p,
```

Comment:
So far mutation_partition was a future-oblivious building block.

Perhaps we should keep it this way and make these apply_gently external utilities in separate files.

Reply:
okay, I'll try refactoring out to a separate file

Reply:
An alternative is to make a generic building block, which can drive the `apply_monotonically()` to completion, i.e. call any method which takes `is_preemptible` and invoke it in a loop, until it finishes, yielding in-between.

Reply:
@denesb we already have that: see apply_monotonically

Reply:
Yes. What I mean is that we can have wrapper which can drive such a method to completion, exposing a `future<>` based interface to the caller. This can live in a separate file, like Avi suggested.

Reply:
I'm adding mutation/async_utils in the next version